### PR TITLE
test(e2e): Generate list of pages to test using Eleventy

### PR DESCRIPTION
**Pull request by jpveooys**
This switches from using a hard-coded list of pages in the end-to-end tests to a list automatically generated by Eleventy.

It's not ideal as it means that `dist/page-list.json` has to exist before `e2e/constants.ts` is valid, but otherwise it works (for `npm start` too).

I also tried to use a dynamic import, but it means you can't parametrise the tests in the normal way (as it needs an `await`). (There is `test.step()`, but that affects parallelism and other things.)

Eleventy also has `--to=json` CLI option (which spits out the entire site as JSON), but it doesn't help much and it's a bit unwieldy as it includes all the page content.

@adamledwards Let me know what you think of this approach.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -3,8 +3,10 @@
     "esModuleInterop": true,
     "forceConsistentCasingInFileNames": true,
     "moduleResolution": "node",
+    "resolveJsonModule": true,
     "strict": true,
     "skipLibCheck": true,
+    "target": "ES2021",
```
Comment (jpveooys) [inline tsconfig.json:9]: Stops a type error using `String.prototype.replaceAll()` in code that runs in Node.

File: .github/workflows/lint-and-test.yml [hunk 2]
```diff
@@ -18,6 +18,10 @@ jobs:
       - name: Set up environment
         uses: ./.github/actions/prepare-environment
 
+      # Ensure dist/page-list.json has been created
```
Comment (adamledwards) [inline .github/workflows/lint-and-test.yml:21]: That's a shame

**Comment by adamledwards** [comment]
PR looks good, I can't think of a better way to be honest

File: src/site/_includes/layouts/markdown-content-page.njk [hunk 3]
```diff
@@ -7,7 +7,7 @@
   <div class="guidance-split-pane__nav">
     {{ guidanceSideNav({
       url: page.url,
-      items: collections[tags[0]] | sort(false, false, 'data.order')
+      items: collections[tags[0]] | rejectattr_path('data.is_root') | sort(false, false, 'data.order')
```
Comment (adamledwards) [inline src/site/_includes/layouts/markdown-content-page.njk:10]: Looks good, I probably need the inverse of this that only includes the root for the primary nav

**Review by adamledwards — APPROVED** [review]

File: src/site/_includes/layouts/markdown-content-page.njk (hunk 3 above)
Comment (jpveooys) [reply src/site/_includes/layouts/markdown-content-page.njk:10]: Yeah I thought I left a comment but the built-in `rejectattr`/`selectattr` don't support dotted paths.

**Pull request merged**